dxfeed/core/DXFeedPy.pyx

```diff
@@ -105,9 +105,7 @@ cdef class SubscriptionClass:
         self.listener = NULL
 
     def __dealloc__(self):
-        if self.subscription:  # if connection is not closed
-            clib.dxf_close_subscription(self.subscription)
-            self.con_sub_list_ptr[0][self.subscription_order] = NULL
+        dxf_close_subscription(self)
```

Comment:
Looks great, but what about handling in {{dxf_close_connection}}? It'll be better to use there same way for each individual sub.

Reply:
new logic with weakref implemented